mkdocs.yml

```diff
@@ -109,13 +109,21 @@ nav:
     - Policy Attachment: reference/policy-attachment.md
   - Enhancements:
     - Overview: geps/overview.md
-    - Provisional:
```

Comment:
My thought process behind the previous ordering was that _most_ people that are viewing enhancements probably want to see the enhancements that are in progress and/or have recently changed, and those are most likely to be provisional or experimental. That led to ascending levels of stability with "declined" moved all the way to the end as that effectively means "end of life" for a GEP. 

This follows Nick's suggestion to change the order to descending levels of stability. Comparing the [deploy preview for this](https://deploy-preview-2704--kubernetes-sigs-gateway-api.netlify.app/geps/overview/) with the [ordering that's live](https://gateway-api.sigs.k8s.io/geps/overview/), I think I may have a very slight preference for the state before this PR, but I don't feel strongly enough to block a change here.

Reply:
I guess my feeling is that we should be emphasizing the things that are more stable by putting them first - right now the Memorandum and Standard sections are not that populated, but over time we expect these to get much more full.

I think that the fact that the API is GA now means that we will have an increasing number of readers of this page that are concerned with the stuff that's done, rather than the stuff that's being worked on.

So I think that putting the more stable things at the top prepares us for that future.

Reply:
I think that leads to an interesting question about the fundamental purpose of GEPs. Should they be used as a reference point for how we made a decision or should they take the place of documentation/API Spec and be a source of truth? I tend to think that GEPs are more to show how we reached a conclusion but that we should consider the following order in terms of authoritativeness:

1. API Spec
2. Conformance Tests
3. Documentation
4. GEP

IE when something is unclear in one, go down the list until it is, but if there's a conflict between a GEP and the API Spec, always trust the API Spec. If we agree with that general idea, I'd say the primary reasons to look at GEPs would be either:

1. Read up on a new provisional or experimental feature that is not well documented yet 
2. Read historical rationale for making certain design decisions in existing API

I'm not sure which of those would be more common, but I'd tend to lean towards the first one, which would make me lean towards the existing ordering. This is not a strong opinion though.

In either case, if we agree on the order of authoritativeness that I just came up with, I can write a follow up PR to document that somewhere.